addons/mrp/wizard/mrp_production_backorder.py

```diff
@@ -31,10 +31,18 @@ def _compute_show_backorder_lines(self):
             wizard.show_backorder_lines = len(wizard.mrp_production_backorder_line_ids) > 1
 
     def action_close_mo(self):
-        return self.mrp_production_ids.with_context(skip_backorder=True, no_procurement=True).button_mark_done()
+        ctx = dict(self.env.context)
+        if ctx.get('workorder_id'):
+            workorder_id = self.env['mrp.workorder'].browse(ctx.get("workorder_id"))
+            return workorder_id.with_context(ctx, skip_backorder=True).record_production()
+        return self.mrp_production_ids.with_context(skip_backorder=True).button_mark_done()
 
     def action_backorder(self):
         ctx = dict(self.env.context)
         ctx.pop('default_mrp_production_ids', None)
         mo_ids_to_backorder = self.mrp_production_backorder_line_ids.filtered(lambda l: l.to_backorder).mrp_production_id.ids
-        return self.mrp_production_ids.with_context(ctx, skip_backorder=True, mo_ids_to_backorder=mo_ids_to_backorder).button_mark_done()
+        if ctx.get('workorder_id'):
+            workorder_id = self.env['mrp.workorder'].browse(ctx.get("workorder_id"))
+            return workorder_id.with_context(ctx, skip_backorder=True, mo_ids_to_backorder=mo_ids_to_backorder).record_production()
+        res = self.mrp_production_ids.with_context(ctx, skip_backorder=True, mo_ids_to_backorder=mo_ids_to_backorder).button_mark_done()
+        return res
```

Comment:
Why? Remove this, you should not record production from here. The MO validation should be responsible to close the open workorders

Reply:
I guess it's due to the wizard now happening in the WO process. Then add a links to workorder_ids and don't use the context